### PR TITLE
fix: increase abci socket message size limit to 2GB

### PR DESCRIPTION
**Pull request by troykessler**
This PR addresses #1729 by increasing the maximum message size for the ABCI socket protocol to 2GB. The current maximum message size of 100MB caused InitChain to fail if the genesis file is bigger than 100MB because the app state is  a parameter in this ABCI method (see [here](https://github.com/cometbft/cometbft/blob/main/spec/abci/abci++_methods.md#initchain)). The new message limit should cover all existing genesis files while still being a protection against large spam messages.

---

#### PR checklist

- [ ] Tests written/updated
- [x] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: .changelog/unreleased/improvements/1730-increase-abci-socket-message-size-limit [hunk 1]
```diff
@@ -0,0 +1 @@
+- `[fix]` Increase ABCI socket message size limit to 2GB ([\#1730](https://github.com/cometbft/cometbft/pull/1730))
```
Comment (melekes) [inline .changelog/unreleased/improvements/1730-increase-abci-socket-message-size-limit:1]:
```suggestion
- `[abci]` Increase ABCI socket message size limit to 2GB ([\#1730](https://github.com/cometbft/cometbft/pull/1730))
```

**Comment by troykessler** [comment]
Changed `maxMsgSize` from `2147483648` to `2147483647` (MaxInt32) to allow successful builds on 32-bit systems where a message size of 2GB (MaxInt32 + 1) resulted in the following error: `cannot use maxMsgSize (untyped int constant 2147483648) as int value in argument to protoio.NewDelimitedReader (overflows)`

File: .changelog/unreleased/improvements/1730-increase-abci-socket-message-size-limit [hunk 2]
```diff
@@ -0,0 +1 @@
+- `[abci]` Increase ABCI socket message size limit to 2GB ([\#1730](https://github.com/cometbft/cometbft/pull/1730))
```
Comment (thanethomson) [inline .changelog/unreleased/improvements/1730-increase-abci-socket-message-size-limit:1]:
```suggestion
- `[abci]` Increase ABCI socket message size limit to 2GB ([\#1730](https://github.com/cometbft/cometbft/pull/1730): @troykessler)
```

**Review by thanethomson — APPROVED** [review]
Thanks @troykessler!

**Pull request merged**